halogen/vnd/error.py

```diff
@@ -0,0 +1,72 @@
+"""vnd.error support."""
+
+import halogen
+import six
+
+
+class Error(Exception):
+    """Base exception."""
+
+    def __init__(self, message, path=None, errors=None):
+        """Create an error.
+
+        :param message: Error message.
+        :param path: Optional JSON Pointer path.
+        :param errors: Optional nested errors.
+        """
+        self.message = message
+        if path is not None:
+            self.path = path
+        if errors is not None:
+            self.errors = errors
+
+    @classmethod
+    def from_validation_exception(cls, exception, **kwargs):
+        """Create an error from validation exception."""
+        errors = []
+
+        def flatten(error, path=""):
+            if isinstance(error, halogen.exceptions.ValidationError):
+                if not path.endswith("/"):
+                    path += "/"
+                if error.attr is not None:
+                    path += error.attr
+                elif error.index is not None:
+                    path += str(error.index)
+
+                for e in error.errors:
+                    flatten(e, path)
+            else:
+                message = error
+                if isinstance(error, Exception):
+                    try:
+                        message = error.message
+                    except AttributeError:
+                        message = six.text_type(error)
+                # TODO: i18n
+                errors.append(Error(message=message, path=path))
+
+        flatten(exception)
+        message = kwargs.pop("message", "Validation error.")
+        return cls(message=message, errors=sorted(errors, key=lambda error: error.path), **kwargs)
```

Comment:
nice hack the ordering based on error.path. Is it possible that path is None? In that case python3 won't allow comparison between None and str

Reply:
or "" will do?